Client/Frontend/Shields/ShareTrayView.swift

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+// MARK: - ShareTrackersViewDelegate
+
+protocol ShareTrayViewDelegate: AnyObject {
+
+    func didShareWithMail(_ view: ShareTrayView)
+    func didShareWithTwitter(_ view: ShareTrayView)
+    func didShareWithFacebook(_ view: ShareTrayView)
+    func didShareWithDefault(_ view: ShareTrayView)
+}
+
+// MARK: - ShareTrackersView
+
+class ShareTrayView: UIView, Themeable {
+    
+    // MARK: Properties
+    
+    private let mailShareButton = UIButton().then {
+        $0.addTarget(self, action: #selector(tappedMailShareButton), for: .touchUpInside)
+        $0.setImage(#imageLiteral(resourceName: "share-bubble-mail").template, for: .normal)
+        $0.contentMode = .scaleAspectFit
+        $0.tintColor = .white
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    }
+    
+    private let twitterShareButton = UIButton().then {
+        $0.setImage(#imageLiteral(resourceName: "share-bubble-twitter").template, for: .normal)
+        $0.contentMode = .scaleAspectFit
+        $0.tintColor = .white
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    }
+    
+    private let facebookShareButton = UIButton().then {
+        $0.setImage(#imageLiteral(resourceName: "share-bubble-facebook").template, for: .normal)
+        $0.contentMode = .scaleAspectFit
+        $0.tintColor = .white
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    }
+    
+    private let defaultShareButton = UIButton().then {
```

Comment:
Maybe create all of these buttons with a helper function and pass the image as a parameter? That way we don't duplicate all of the code and they still look the same?

Reply:
Fix at https://github.com/brave/brave-ios/pull/3151/commits/a0bbdc866aaf083a6047d7fe44e4acd16c4826f5

Reply:
@soner-yuksel 

I mean buttons :) but still you made a great change.
See below though for what I mean:

```swift
private let someButton = UIButton().then {
        $0.setImage(#imageLiteral(resourceName: someImageName).template, for: .normal)
        $0.contentMode = .scaleAspectFit
        $0.tintColor = .white
        $0.setContentHuggingPriority(.required, for: .horizontal)
        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
    }
    ```
    
    Right now, 4 buttons share the same code with only the image being different? If we put that code into a function and do: `let facebookShareButton = someFunction()`, `let mailShareButton = someFunction()`, etc? Will it help?

Reply:
Understood :) fix at 01f4b5f